### ISSUE TITLE
Validate at entry in addition to submission

### ISSUE THREAD

**Issue by fgregg** [datamade/large-lots#170]
Many of the validations steps could be done when someone is done filing out the field, not just when clicking submit. I understand that this is a bigger issue which we probably won't get to before launch.

**Comment by derekeder** [comment]
Doing it this way would make the validation 100% javascript. We'd still have to validate the way we are currently doing it.

**Comment by fgregg** [comment]
Yes. the javascript validation couldn't replace the server side validation.
would make for better form experience, though.
